modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCDataStore.java

```diff
@@ -3834,7 +3834,7 @@ protected PreparedStatement selectBoundsSQLPS(
                             "AND " + toSQL.encodeToString(filter));
                     toSQL.setInline(false);
                 } else {
-                    sql.append(toSQL.encodeToString(filter));
+                    sql.append(" ").append(toSQL.encodeToString(filter));
```

Comment:
So this restores the situation before the placeholder changes, for the record:

````
            try {
                toSQL = createPreparedFilterToSQL(featureType);
                sql.append(" ").append(toSQL.encodeToString(filter));
            } catch (FilterToSQLException e) {
                throw new RuntimeException(e);
            }
````